plutus-tx/src/PlutusTx/Builtins/Internal.hs

```diff
@@ -197,6 +199,17 @@ verifySignature :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString -
 verifySignature (BuiltinByteString pubKey) (BuiltinByteString message) (BuiltinByteString signature) =
   coerce (fromMaybe False (Crypto.verifySignature pubKey message signature))
 
+{-# NOINLINE verifySECP256k1Signature #-}
+verifySECP256k1Signature :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString -> BuiltinBool
+verifySECP256k1Signature (BuiltinByteString pk) (BuiltinByteString sig) (BuiltinByteString msg) =
+  case Crypto.verifySECP256k1Signature pk sig msg of
+    Emitter f -> case runIdentity (f go) of
+      EvaluationFailure   -> mustBeReplaced "SECP256k1 verification errored."
+      EvaluationSuccess b -> BuiltinBool b
+  where
+    go :: Text -> Identity ()
+    go t = Identity . trace (BuiltinString t) $ ()
```

Comment:
Have you tried running it to see if it indeed traces something?

The monad instance for `Identity` is defined as

```haskell
instance Monad Identity where
    m >>= k  = k (runIdentity m)
```

i.e. impure effects in `m` are not performed unless `k` forces its arguments, which we may or may not do.

What if you just made `go` run in `EvaluationResult` rather than `Identity`? Something like

```haskell
go :: Text -> EvaluationResult ()
go t = trace (BuiltinString t) $ EvaluationSuccess ()
``` 

(note that `trace` is outside of `EvaluationSuccess` rather than inside, because `EvaluationResult` is not strict and so similar considerations apply)

Reply:
I wasn't sure about the right way to do this, so if you think it should be using `trace` like that, I'll change it.